### PR TITLE
First last page

### LINKED ISSUE #5: Create end page and first page buttons with optional props

### PR DESCRIPTION
**Pull request by aelamel**
Closes #5

### CODE REVIEW

File: src/ReactPaginationNav.js [hunk 1]
```diff
@@ -30,6 +34,16 @@ const ReactPaginationNav = ({
 
   return (
     <div className={cn('react-pagination-nav', `react-pagination-nav--${theme}`, className)}>
+      {!isFirstBtnHidden && (
+        <button
+          className="react-pagination-nav__first-page react-pagination-nav__button"
+          onClick={() => goToFirstPage()}
```
Comment (mohitkyadav) [inline src/ReactPaginationNav.js]: add it to prop types and double check
`goToFirstPage && goToFirstPage()`

File: src/ReactPaginationNav.js [hunk 2]
```diff
@@ -76,6 +90,17 @@ const ReactPaginationNav = ({
           {'>'}
         </button>
       )}
+
+      {!isLastBtnHidden && (
+        <button
+          className="react-pagination-nav__last-page react-pagination-nav__button"
+          onClick={() => goToLastPage()}
```
Comment (mohitkyadav) [inline src/ReactPaginationNav.js]: add it to prop types and double check
`goToLastPage && goToLastPage()`

**Review by mohitkyadav — CHANGES_REQUESTED** [review]

File: src/ReactPaginationNav.js (hunk 1 above)
Comment (aelamel) [reply src/ReactPaginationNav.js]: Done

**Review by aelamel — COMMENTED** [review]
Changes have been made

**Review by mohitkyadav — APPROVED** [review]

**Comment by mohitkyadav** [comment]
please resolve conflicts @aelamel

**Review by mohitkyadav — APPROVED** [review]
conflicts @aelamel and LGTM

File: src/ReactPaginationNav.js [hunk 3]
```diff
@@ -22,6 +26,9 @@ const ReactPaginationNav = ({
   className, goToPreviousPage, pageCount, currentPage,
   goToPage, goToNextPage, visiblePages = 5,
   isPreviousBtnHidden, isNextBtnHidden,
+  goToFirstPage, goToLastPage,
+  isFirstBtnHidden, isLastBtnHidden,
+  PageButton,
```
Comment (tklg) [inline src/ReactPaginationNav.js:31]: Duplicate prop PageButton

```suggestion
```

File: src/ReactPaginationNav.js [hunk 4]
```diff
@@ -32,6 +39,26 @@ const ReactPaginationNav = ({
...
+        <button
+          className="react-pagination-nav__first-page react-pagination-nav__button"
+          onClick={() => goToFirstPage && goToFirstPage()}
+          title="Go to first page"
+          aria-label="Go to first page"
+        >
+          {'First'}
+        </button>
+      )}
+      {!isPreviousBtnHidden && (
```
Comment (tklg) [inline src/ReactPaginationNav.js:52]: Previous button should be handled by PrevNextButton prop, remove duplicate

File: src/ReactPaginationNav.js [hunk 5]
```diff
@@ -45,11 +72,42 @@ const ReactPaginationNav = ({
                   (Math.abs(currentPage - 1 - i) < (oddVisiblePages - (pageCount - currentPage)))
                 )
               ) &&
+              <button
```
Comment (tklg) [inline src/ReactPaginationNav.js:75]: handled by PageButton prop, remove duplicate

File: src/ReactPaginationNav.js [hunk 6]
```diff
@@ -45,11 +72,42 @@ const ReactPaginationNav = ({
...
+                onClick={() => goToPage(i + 1)}
+              >
+                {i + 1}
+              </button>
               <PageButton page={i + 1} active={currentPage === i + 1} onClick={() => goToPage(i + 1)} />
             )
           })
         }
       </div>
+      {!isNextBtnHidden && (
```
Comment (tklg) [inline src/ReactPaginationNav.js:90]: handled by PrevNextButton prop, remove duplicate and move lastBtn after the nextBtn

**Comment by aelamel** [comment]
Fixed in #18

**Pull request merged**